src/components/home.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import JSONTree from 'react-json-tree';
+
+import DivJSON from './div_json';
+
+import { createAuth, getTokens, renewIdToken, decodeIdToken, refreshSession, closeSession, signOut } from '../actions/index';
+
+class Home extends Component {
+  componentWillMount() {
+    //  create auth that goes in props
+    this.props.createAuth();
+  }
+
+  sessionClosed() {
+    return localStorage.getItem('session_closed');
+  }
+
+  getTokens() {
+    const { auth, login } = this.props;
+    this.props.getTokens(login, auth);
+  }
+
+  renewIdToken() {
+    const { auth } = this.props;
+    this.props.renewIdToken(auth);
+  }
+
+  decodeIdToken() {
+    const { auth, tokens } = this.props;
+    this.props.decodeIdToken(tokens.idToken, auth);
+  }
+
+  refreshSession() {
+    const { auth } = this.props;
+    this.props.refreshSession(auth);
+  }
+
+  closeSession() {
+    const { auth } = this.props;
+    localStorage.setItem('session_closed', true);
+    this.props.closeSession(auth);
+  }
+
+  signOut() {
+    const { auth } = this.props;
+    this.props.signOut(auth);
+  }
+
+  clearStorageAnRedirect() {
+    localStorage.clear();
+    window.location = "/";
+  }
+
+  render() {
+    const { login, session, tokens } = this.props;
+
+    if(session.logout) {
+      this.clearStorageAnRedirect();
+    }
+    
+    if(login && login.success) {
+      return (
+        <div className="row under-header"> 
+          <div className="col-md-2"><br />
+            <div className="affix">
+              { (() => {
+                  if(session.active && !this.sessionClosed()) {
```

Comment:
For these kinds of things - I'd probably create helper functions that tell me boolean results, vs. doing the javascript directly in the JSX.